### PR TITLE
[doc][orc8r] Add documentation on AWS resources deployed for Orchestrator

### PR DESCRIPTION
**Pull request by joary**
## Summary

- Added documentation on Orc8r resources deployed on AWS
- Included documentation in docusaurus sidebar

## Test Plan

image

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by hcgatewood — CHANGES_REQUESTED** [review]
Hey Joary thanks for adding this. Couple requests

High level
- In the test plan, include a screen cap of the page running locally. At minimum so we can see the included image
- Let's make a new "Architecture" subsection under "Orchestrator", i.e. `Orchestrator > Architecture`
- Then we can move your AWS info as well as this page https://magma.github.io/magma/docs/next/orc8r/intro into that section, which feels like a better home

Particular: some small nits for the documentation style
- Capitalization: make a pass to ensure consistency for the following capitalizations
    - K8s, Kubernetes
    - Orc8r, Orchestrator
    - NMS
    - Magma
    - Helm
- Don't use "here" as hyperlink text. The [chosen hyperlink text should be self-descriptive](https://www.nngroup.com/articles/writing-links/)

**Review by hcgatewood — CHANGES_REQUESTED** [review]
Hey @joary, the Docusaurus layout was updated recently, so you'll need to rebase

Aside from that, content looks good. Consider taking a look at Magma Markdown conventions as part of your last-pass over the PR https://magma.github.io/magma/docs/next/contributing/contribute_conventions#markdown

Looking forward to merging this in!

**Comment by hcgatewood** [comment]
Ping @joary

File: docs/docusaurus/i18n/en.json [hunk 1]
```diff
@@ -283,6 +283,9 @@
       "orc8r/deploy_orcl": {
         "title": "Experimental Orc8r Deployer"
       },
+      "orc8r/dev_aws_stack": {
+        "title": "Orchestrator AWS Stack"
```
Comment (hcgatewood) [inline docs/docusaurus/i18n/en.json]:
```suggestion
        "title": "AWS Stack"
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+---
+id: dev_aws_stack
+title: Orchestrator AWS Stack
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
title: AWS Stack
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+hide_title: true
+---
+
+# Orchestrator Stack on AWS
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
# AWS Stack
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md:10]:
```suggestion
This document describes the Orchestrator stack when it's deployed to AWS.

```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+![orc8r_aws_stack](assets/orc8r/orc8r_aws_stack.png)
+
+## What does terraform deploy in my AWS account?
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
## What does Terraform deploy in my AWS account?
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+The following resources are the basic infrastructure necessary to deploy Magma services:
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
The following resources are the basic infrastructure necessary to deploy Magma services
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+* **Kubernetes cluster:** Orc8r runs over a Kubernetes (K8s) cluster, Magma uses Elastic Kubernetes Service (EKS) to deploy the cluster. Other resources are created to support this cluster.
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
* **Kubernetes cluster (EKS).** Orc8r runs over a Kubernetes (K8s) cluster, Magma uses Elastic Kubernetes Service to deploy the cluster. Other resources are created to support this cluster.
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+* **Elastic File System (EFS):** File system to hold persistent data for Kubernetes cluster.
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]: Change all these colons to periods

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+|Name	|Type	|Description	|Default Value	|
+|---	|---	|---	|---	|
+|orc8r_db_password	|Mandatory	|Password for the Orchestrator DB. Must be at least 8 characters.	|-	|
+|orc8r_db_storage_gb	|Optional	|Capacity in GB to allocate for Orchestrator RDS instance.	|64	|
+|orc8r_db_engine_version	|Optional	|Postgres engine version for Orchestrator DB.	|9.6.15	|
+|cluster_version	|Optional	|Kubernetes version for the EKS cluster.	|1.17	|
+|elasticsearch_version	|Optional	|ES version for ES domain.	|7.1	|
+
+The values for these configurations can be defined/overwritten in your main.tf `orc8r` module.
+Some configurations do not have default values and are mandatory to be present in your main.tf.
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
Some configurations do not have default values and are mandatory to be present in your `main.tf`.
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+
+|Name	|Type	|Description	|Default Value	|
+|---	|---	|---	|---	|
+|orc8r_db_password	|Mandatory	|Password for the Orchestrator DB. Must be at least 8 characters.	|-	|
+|orc8r_db_storage_gb	|Optional	|Capacity in GB to allocate for Orchestrator RDS instance.	|64	|
+|orc8r_db_engine_version	|Optional	|Postgres engine version for Orchestrator DB.	|9.6.15	|
+|cluster_version	|Optional	|Kubernetes version for the EKS cluster.	|1.17	|
+|elasticsearch_version	|Optional	|ES version for ES domain.	|7.1	|
+
+The values for these configurations can be defined/overwritten in your main.tf `orc8r` module.
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
The values for these configurations can be defined/overwritten in your `main.tf` `orc8r` module.
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+
+|Name	|Type	|Description	|Default Value	|
+|---	|---	|---	|---	|
+|region	|Mandatory	|AWS region to deploy Orchestrator components into. The chosen region must provide EKS.	|-	|
+|orc8r_domain_name	|Mandatory	|Base domain name for Orchestrator.	|-	|
+|orc8r_deployment_type	|Mandatory	|Type of Orc8r deployment (fixed wireless access, federated fixed wireless access, or all modules)	|-	|
+|orc8r_tag	|Mandatory	|Image tag for Orchestrator components.	|-	|
+|deploy_openvpn	|Optional	|Flag to deploy OpenVPN server into cluster. This is useful if you want to remotely access AGWs.	|FALSE	|
+
+The values of these configurations can be defined/overwritten in your main.tf `orc8r-app` module.
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
The values of these configurations can be defined or overwritten in your `main.tf` file's `orc8r-app` module.
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+## What does the Orc8r Helm chart contain?
+
+
+Orc8r Helm chart is composed of 5 other charts:
+
+1. **Logging:** deploys all the services related to collect and store logs, it includes:
+    1. FluentD: for log aggregation
+    2. Elastic Search Curator: to clean old logs
+2. **Metrics:** deploys all the services related to collect, store and show metrics, it includes:
+    1. Prometheus/Thanos for metric storage.
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
    1. Prometheus for metric storage.
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+    1. Nginx HTTP proxy
+    2. NMS service
+    3. [More details](https://github.com/magma/magma/blob/master/orc8r/cloud/helm/orc8r/charts/nms/README.md)
+4. **Orc8r:** deploys the Magma orchestration function
+    1. Orc8r controller: Stream configuration to all AWGs in the network. Collect and export metrics from all the AGWs in the network.
+    2. [More details](https://github.com/magma/magma/blob/master/orc8r/cloud/helm/README.md)
+5. **Secrets:** is used to apply a set of secrets required by the Magma Orchestrator.
+    1. [More details](https://github.com/magma/magma/blob/master/orc8r/cloud/helm/orc8r/charts/secrets/README.md)
+
+## Which certificates Magma use?
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
## Which certificates does Magma use?
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+
+Orc8r needs certificates to assure messages traveling over the internet are encrypted and secure. Certificates are created in pairs with a public certificate (.crt or .pem) and a private key (.key). The former needs to be safely stored and kept secret, while the latter can be distributed to clients. Please [read the certificates section on Orchestrator architecture](https://magma.github.io/magma/docs/next/orc8r/dev_security#certificates).
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
Orc8r needs certificates to assure messages traveling over the internet are encrypted and secure. Certificates are created in pairs with a public certificate (`.crt` or `.pem`) and a private key (`.key`). The former needs to be safely stored and kept secret, while the latter can be distributed to clients. Please read the [certificates section on Orchestrator architecture](https://magma.github.io/magma/docs/next/orc8r/dev_security#certificates).
```

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+
+
+<span style="color:red">
+**Attention:** All the certificates are created with validity time period, so make sure you know when your certificates expire and schedule a maintenance to update them.
+</span>
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
> ***NOTE:*** All certificates are created with validity time period, so make sure you know when your certificates expire and schedule a maintenance to update them.
```

**Review by hcgatewood — APPROVED** [review]
Great thanks Joary. Accepted with a couple comments, please incorporate before merging 🙂

**Comment by joary** [comment]
@hcgatewood I included all your comments, though I'm not authorized to merge.

File: docs/readmes/orc8r/dev_aws_stack.md [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+
+This document describes the Orchestrator stack when it's deployed to AWS.
+
+![orc8r_aws_stack](assets/orc8r/orc8r_aws_stack.png)
+
+## What does Terraform deploy in my AWS account?
+
+The following resources are the basic infrastructure necessary to deploy Magma services
+
+* **Kubernetes cluster (EKS).* Orc8r runs over a Kubernetes (K8s) cluster, Magma uses Elastic Kubernetes Service to deploy the cluster. Other resources are created to support this cluster.
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_aws_stack.md]:
```suggestion
* **Kubernetes cluster (EKS).** Orc8r runs over a Kubernetes (K8s) cluster, Magma uses Elastic Kubernetes Service to deploy the cluster. Other resources are created to support this cluster.
```

**Review by hcgatewood — APPROVED** [review]
One last nit (formatting error) then I'll merge for you!

**Comment by joary** [comment]
Thanks @hcgatewood, I solved it.

**Comment by joary** [comment]
@hcgatewood , I had to rebase last two commits to include sign, DCO was failing for those.

**Pull request merged**